server/raft.go

```diff
@@ -1134,7 +1144,7 @@ func (n *raft) InstallSnapshot(data []byte) error {
 
 	// Check that a catchup isn't already taking place. If it is then we won't
 	// allow installing snapshots until it is done.
-	if len(n.progress) > 0 || n.paused {
+	if len(n.progress) > 0 || n.inflightSnapshot {
```

Comment:
The `n.paused` was added in https://github.com/nats-io/nats-server/pull/6053 and is a fix that went into 2.10.x
However `n.inflightSnapshot` is a more correct approach, as it blocks installing a snapshot even if the upper layer didn't receive the snapshot or didn't call `n.PauseApply()` yet. So this is a more solid fix for 2.11